src/ts/CmcdIntegration.ts

```diff
@@ -120,9 +120,12 @@ export class CmcdIntegration {
     const data = this.gatherCmcdData(type, request);
 
     if (this.useQueryArgs) {
+      const url = new URL(request.url);
+      url.searchParams.delete('CMCD');
+
       const cmcdStr = cmcdDataToUrlParameter(data);
-      const separator = request.url.includes('?') ? '&' : '?';
-      request.url = `${request.url}${separator}${cmcdStr}`;
+      const separator = url.toString().includes('?') ? '&' : '?';
+      request.url = `${url.toString()}${separator}${cmcdStr}`;
```

Comment:
Now that we already have an `URL` object, we could also use the `url.searchParams.append` method (but it would need further changes as the `cmcdDataToUrlParameter` method already returns a key-value string.
But also works as is of course.